### PR TITLE
FIX: add mingw_py27 patch

### PR DESCRIPTION
**Pull request by kmuehlbauer**
Adds a patch to get extensions compile on mingw, see https://github.com/numpy/numpy/pull/9280

ping @ocefpaf

### CODE REVIEW

**Comment by ocefpaf** [comment]
Thanks @kmuehlbauer this won't affect the feedstock here but will help me a lot in my local builds of Fortran extensions.

I will check to see if we can enable the Windows on conda-forge builds using a different blas somehow...

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,6 +10,8 @@ source:
   fn: numpy-{{ version }}.tar.gz
   url: https://github.com/numpy/numpy/archive/v{{ version }}.tar.gz
   sha256: dbbaaf3e86d0d138d42c61b1243d580bd6351088f0bcf2e44a7374c4559a1845
+  patches:
+    - mingw_py27.patch
```
Comment (ocefpaf) [inline recipe/meta.yaml]: @kmuehlbauer can you add a comment and a link to the upstream PR so we can track this?

**Comment by ocefpaf** [comment]
@mingwandroid and @msarahan it would be nice if this patch could made it into `defaults`. That would allow us to build Fortran extensions, like https://github.com/conda-forge/pytmatrix-feedstock/issues/8, on Windows + numpy 1.13

**Comment by mingwandroid** [comment]
Hah, @diorcety (the original reporter - Hey Yann) is a friend of mine actually, we worked on crosstool-ng together but not Python stuff yet; small world.

@ocefpaf, please take a look at using `m2w64-openblas`, I've used it successfully as has @SylvainCorlay. BTW is there a good reason for `conda-forge` not to use `MKL` these days?

**Comment by ocefpaf** [comment]
> Hah, @diorcety (the original reporter - Hey Yann) is a friend of mine actually, we worked on crosstool-ng together but not Python stuff yet; small world.

Full circle :smile: 

> @ocefpaf, please take a look at using `m2w64-openblas`, I've used it successfully as has @SylvainCorlay.

Awesome! I will take a look soon.

> BTW is there a good reason for `conda-forge` not to use `MKL` these days?

IMO, but I don't really have the energy to discuss that again :unamused:

**Comment by diorcety** [comment]
@mingwandroid Small world :D

**Comment by rolk** [comment]
I just wanted to let you know that I added an improvement to numpy/numpy#9280 which I think will make it work better in virtualenvs in Python 2.7 (and I think it is fairly common to use virtualenvs to custom-compile Fortran extensions); you may want to pick this up as well when/if the original pull request is merged.

**Pull request merged**